blocks/library/table/table-block.js

```diff
@@ -3,9 +3,28 @@ import BlockControls from '../../block-controls';
 import { Toolbar, DropdownMenu } from 'components';
 import { __ } from 'i18n';
 
+function isTableSelected( editor ) {
+	return editor.dom.getParent(
+		editor.selection.getStart( true ),
+		'table'
```

Comment:
Turned out that it needed to be the parent of the editor body otherwise it would not consider the table. Otherwise I think this issue is fixed.